sites/bestbuy.py

```diff
@@ -6,327 +6,563 @@
-                r = requests.get("https://www.bestbuy.com/api/csiservice/v2/key/tas",headers=headers)
-                self.status_signal.emit({"msg":"Got TAS Data","status":"normal"})
+                self.status_signal.emit({"msg": "Getting TAS Data", "status": "normal"})
+                r = requests.get("https://www.bestbuy.com/api/csiservice/v2/key/tas", headers=headers)
+                self.status_signal.emit({"msg": "Got TAS Data", "status": "normal"})
                 return json.loads(r.text)
             except Exception as e:
-                self.status_signal.emit({"msg":"Error Getting TAS Data(line {} {} {})".format(sys.exc_info()[-1].tb_lineno, type(e).__name__, e),"status":"error"})
+                self.status_signal.emit({"msg": "Error Getting TAS Data(line {} {} {})".format(
+                    sys.exc_info()[-1].tb_lineno, type(e).__name__, e), "status": "error"})
                 time.sleep(self.error_delay)
+
     def monitor(self):
         headers = {
-            "accept": "text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9",
-            "accept-encoding": "gzip, deflate, br",
-            "accept-language": "en-US,en;q=0.9,zh-CN;q=0.8,zh;q=0.7",
-            "cache-control": "max-age=0",
-            "upgrade-insecure-requests": "1",
-            "user-agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.69 Safari/537.36"
+            "Accept": "text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8",
+            "Accept-Encoding": "gzip, deflate",
+            "Accept-Language": "en-US,en;q=0.5",
+            "Upgrade-Insecure-Requests": "1",
+            "user-agent": settings.userAgent
         }
         image_found = False
-        product_image = ""
         while True:
-            self.status_signal.emit({"msg":"Loading Product Page","status":"normal"})
+            self.status_signal.emit({"msg": "Loading Product Page", "status": "normal"})
             try:
-                r = self.session.get(self.product,headers=headers)
+                r = self.session.get(self.product, headers=headers, verify=False)
                 if r.status_code == 200:
                     doc = lxml.html.fromstring(r.text)
                     if not image_found:
-                        self.sku_id = doc.xpath('//span[@class="product-data-value body-copy"]/text()')[1].strip()
                         product_image = doc.xpath('//img[@class="primary-image"]/@src')[0]
                         self.image_signal.emit(product_image)
                         image_found = True
-                    if self.check_stock():
                         return product_image
-                    self.status_signal.emit({"msg":"Waiting For Restock","status":"normal"})
+                    self.status_signal.emit({"msg": "Waiting For Restock", "status": "normal"})
```

Comment:
Logic bug. This doesn't execute properly.

Reply:
True I added condition check for this.